api/db/getStatus.js

```diff
@@ -52,27 +89,71 @@ function _createClient(bag, next) {
     global.config.dbPort,
     global.config.dbName);
 
-  var testClient = new pg.Client(connectionString);
-  testClient.connect(
+  bag.testClient = new pg.Client(connectionString);
+  bag.testClient.connect(
     function (err) {
       if (err) {
         logger.error('Error while connecting to database. ' +
           util.inspect(err));
         bag.resBody.isReachable = false;
         bag.resBody.error = util.inspect(err);
+      } else {
+        bag.resBody.isReachable = true;
       }
       return next();
     }
   );
 }
 
-function _getStatus(bag, next) {
-  if (!bag.resBody.isReachable) return next();
+function _checkUptime(bag, next) {
+  var who = bag.who + '|' + _checkUptime.name;
+  logger.verbose(who, 'Inside');
+  var pgEnvs = {
+    'PGHOST': global.config.dbHost,
+    'PGPORT': global.config.dbPort,
+    'PGDATABASE': global.config.dbName,
+    'PGUSER': global.config.dbUsername,
+    'PGPASSWORD': global.config.dbPassword
+  };
 
+  var exec = spawn('/usr/bin/psql',
+    ['-t', '-c', 'SELECT now() - pg_postmaster_start_time();'],
+    {
+      env: pgEnvs
+    }
+  );
+
+  exec.stdout.on('data',
+    function (data)  {
+      bag.resBody.uptime = data.toString();
+    }
+  );
+
+  exec.stderr.on('data',
+    function (data)  {
+      bag.resBody.error = data.toString();
```

Comment:
should we error or out `bag.resBody.error` is not empty?

Reply:
if this shows error, it'll be probably not be because db is down(we've already checked that in the previous step). so the error should ideally be propagated to the calller